rpmbuild/copr_rpmbuild/automation/srpm_results.py

```diff
@@ -0,0 +1,39 @@
+"""
+Create `results.json` file for SRPM builds
+"""
+
+import os
+import simplejson
+from copr_rpmbuild.automation.base import AutomationTool
+from copr_rpmbuild.helpers import locate_srpm, get_rpm_header
+
+
+class SRPMResults(AutomationTool):
+    """
+    Create `results.json` for SRPM builds
+    """
+
+    @property
+    def enabled(self):
+        """
+        Do this for every RPM build
+        """
+        return not self.chroot
+
+    def run(self):
+        """
+        Create `results.json`
+        """
+        data = self.get_package_info()
+        path = os.path.join(self.resultdir, "results.json")
+        with open(path, "w", encoding="utf-8") as dst:
+            simplejson.dump(data, dst, indent=4)
+
+    def get_package_info(self):
+        """
+        Return ``dict`` with interesting package metadata
+        """
+        srpm = locate_srpm(self.resultdir)
+        hdr = get_rpm_header(srpm)
+        keys = ["name", "exclusivearch", "excludearch"]
+        return {key: hdr[key] for key in keys}
```

Comment:
I know we agreed on using `python3-specfile` but I tried it and couldn't easily get the `exclusivearch` from it so I used `python3-rpm` instead. We already use it on some places, so it was trivial to use here.

IMHO there is no advantage in using one over the other, so I would just pick whatever is easiest to use. But if we really want `specfile`, I can try to figure out how to use it.

Reply:
Seems like OK for the first step at least.  specfile could resolve some some issues like

```
%if 0%{?fedora} > 39
ExcludeEarch: x86_64
%endif
```

Now thinking again, are such spec files actually resolvable on the single builder that did source build?

Reply:
@nforro  can you provide a guidance to @FrostyX please?

Reply:
> IMHO there is no advantage in using one over the other, so I would just pick whatever is easiest to use.

With `specfile` you can get `ExclusiveArch` like this:

```python
spec = Specfile(...)
with spec.tags() as tags:
    try:
        exclusivearch = tags.exclusivearch.value
    except AttributeError:
        exclusivearch = None
```

or (to get all of them):

```python
spec = Specfile(...)
with spec.tags() as tags:
    exclusivearches = [t.value for t in tags if t.normalized_name == "Exclusivearch"]
```

> IMHO there is no advantage in using one over the other, so I would just pick whatever is easiest to use.

There is a difference, with `specfile` you can access all tags, even those that are enclosed in conditions that would resolve to false on the system where the code is running, while `rpm` gives you only resolved tags.

BTW, you can use `specfile` and still access the underlying `rpm.spec` object with `Specfile.rpm_spec` property.

Reply:
Can I affect the conditions, to get the answer to question like "What is the ExclusiveArch value for Fedora 38+"?

Reply:
Actually, you probably want to use `expanded_value` rather than `value` (as that's what `rpm` would give you).

Reply:
Yes, `expanded_value` is fine, but that doesn't answer the question; we want `expanded_value` on specific distribution (e.g. RHEL 9).  Can we use python-specfile for this?

Reply:
Yes. You can specify macros you want to be defined before every expansion (you can't delete macros loaded from macro files, but you can redefine them):

```python
spec = Specfile("python-specfile.spec", macros=[("fedora", "39"), ("rhel", "")])

# explicitly specify parsed preamble section here, so that macros and conditions are resolved
# be aware though that if you make any changes the original section will be overwritten
with spec.tags(spec.parsed_sections.package) as tags:
    try:
        exclusivearch = tags.exclusivearch.expanded_value
    except AttributeError:
        exclusivearch = None

# for read-only access it's probably better to avoid the context manager
tags = spec.tags(spec.parsed_sections.package).content
try:
    exclusivearch = tags.exclusivearch.expanded_value
except AttributeError:
    exclusivearch = None
```

You can do the same thing with `rpm`, but it would be rather cumbersome.

Reply:
This way of working with spec files is quite revolutionary, thanks.

Reply:
Thank you for the code example, worked for me perfectly.
I updated the PR.

Reply:
I submitted a documentation request https://github.com/packit/specfile/issues/243